Add `nanopet` architecture (#388)
metatensor/metatrain: docs/src/architectures/nanopet.rst

.. _architecture-nanopet:

> NanoPET
> =======
> 
> .. warning::
> 
>   This is an **experimental model**.  You should not use it for anything important.
> 
> This is a more reader- and user-friendly implementation of PET, with slightly
> improved training and evaluation speed.

Reviewer: ```suggestion
NanoPET
=======

.. warning::

  This is an **experimental model**.  You should not use it for anything important.

This is a re-implementation of the original PET (which lives in https://github.com/spozdn/pet), with slightly improved training and evaluation speed.
```

Author: I think the reader-friendlyness is the most important bit actually. If one wants to look into PET's code to understand the architecture, nanoPET is the place to go

Reviewer: maybe, but metatrain users don't care about that =D
